crates/codegen/parser/runtime/Cargo.toml

```diff
@@ -5,11 +5,7 @@ rust-version.workspace = true
 edition.workspace = true
 publish = false
 
-[lib]
```

Comment:
> for example, we never build codegen_parser_runtime into cdylib but only use the source inside

I suggest keeping this. While we don't build the NAPI code here for shipping, we do compile the NAPI types when we compile this crate, so I want to make sure `cdylib` is included, as this affects `cargo check`, and we want to catch any issues early on before codegen runs.

Reply:
The argument sounds reasonable and this probably won't move the needle in the entire pipeline, so I removed the `napi/noop` feature here.

However, adding `cdylib` crate type to anything but `slang_solidity_node_addon`/`solidity_npm_crate` in our case is actively misleading and actually unused.

The `napi-build` configures the linker for dynamic linkage when compiling the crate as cdylib, as we want to be dynamically linked to a Node.js host and the `cdylib` target is only ever used by the `napi build` CLI when building the crate into an addon. We never want to separately compile a `.node` add-on for this stub crate, so this setup is unnecessary and unused here.

It's worth noting that the NAPI macros will still be expanded and checked both in isolation for this crate and for the dependents, so we still run the relevant checks; it's just that we never `napi build` this crate (and shouldn't), so we don't need the setup for it.

Reply:
> It's worth noting that the NAPI macros will still be expanded and checked both in isolation for this crate and for the dependents, so we still run the relevant checks; it's just that we never napi build this crate (and shouldn't), so we don't need the setup for it.

Sounds good. Thanks!

Reply:
I think there is still one mention of `noop` in `crates/solidity/outputs/npm/crate/Cargo.toml`.

Reply:
There is but the paragraph there outlines the general cdylib support story and what we could in theory do, i.e. `napi/noop` for for a hypothetical rlib/cdylib combined crate when not building the node addon but can't, since we run everything with `--all-features` (and Cargo features are additive); specifically it doesn't refer to the now removed `napi/noop` from the parser codegen crate.